lib/process_settings/abstract_monitor.rb

```diff
@@ -11,6 +11,8 @@ class SettingsPathNotFound < StandardError; end
   OnChangeDeprecation = ActiveSupport::Deprecation.new('1.0', 'ProcessSettings::Monitor')
 
   class AbstractMonitor
+    @@full_context_cache = {}
```

Comment:
Let's make this an instance variable rather than a class variable. That way the cache is scoped to a single instance of the Monitor.